blog/2023-07-23-openhab-4-0-release.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: BlogPost
+category: blog
+title: openHAB 4.0 Release
+author: Kai Kreuzer and others
+date: '2023-07-23T18:00:00+02:00'
+previewimage: /uploads/summer23.jpg
+tags:
+  - news
+excerpt: >-
+  A new major release has arrived! openHAB 4 brings some exciting new features and several improvements for our users as well as our devs!
+---
+A new major release has arrived! openHAB 4 brings some exciting new features and several improvements for our users as well as our devs!
+
+We finally reached feature parity between UI-based and file-based configurations with UI support for both transformation and persistence configuration, and moved to Java 17 to give our devs access to great new language features. 
```

Comment:
Perhaps we should also mention a few other benefits from upgrading to Java 17? For example:
- Security enhancements.
- Performance improvements (and reduced memory consumption).
- Making sure we can upgrade external libraries and frameworks we depend upon.

So it's basically about staying current with the latest technology. Maybe something like "staying in sync with the evolving ecosystem". ChatGPT can help. 🙂 I just feel we need to justify and highlight this upgrade a bit more than merely for providing newer language features to the developers. Just a sentence or two.

In the 3.0 blog post we had: 
> Moving from Java 8 to Java 11 brings the language features on a current level and ensures that openHAB continues to run on JVMs that receive security patches.

Reply:
Thanks for your thoughts, I now have:

> As quite usual for new openHAB major releases, we also upgraded to Java 17 to stay in sync with the evolving Java ecosystem. Java 17 brings a wealth of benefits to our user and especially our developers, including security enhancements, performance improvements and reduced memory usage as well as new language features and APIs.

See https://github.com/openhab/website/commit/8bd177db548f3dffab7897daa287f3a415128cf7.